### PR TITLE
feat: add run_binary with output directory support & improved makevar expansion

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: lib/private/run_binary.bzl [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                  " directory named the same as the target.",
+        ),
+        "outs": attr.output_list(
+            doc = "Output files generated by the action.<br/><br/>These labels are available for" +
+                  " <code>$(location)</code> expansion in <code>args</code> and <code>env</code>.",
+        ),
+        "args": attr.string_list(
+            doc = "Command line arguments of the binary.<br/><br/>Subject to" +
+                  "<code><a href=\"https://docs.bazel.build/versions/main/be/make-variables.html#location\">$(location)</a></code>" +
+                  " expansion.",
```
Comment (kormide) [inline lib/private/run_binary.bzl:100]: The html doesn't get rendered in the git preview. Is it supposed to, or is this just for the docsite?

**Review by kormide — APPROVED** [review]

File: docs/copy_file.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+A rule that copies a file to another place.
+
+native.genrule() is sometimes used to copy files (often wishing to rename them).
+The 'copy_file' rule does this with a simpler interface than genrule.
+
+The rule uses a Bash command on Linux/macOS/non-Windows, and a cmd.exe command
+on Windows (no Bash is required).
+
+This fork of bazel-skylib's copy_file adds directory support.
```
Comment (alexeagle) [inline docs/copy_file.md:11]: there was a response on https://github.com/bazelbuild/bazel-skylib/pull/323

I think we should go with `copy_directory` in place of `copy_file(output_dir=True)` so there's some chance of going upstream

File: lib/private/run_binary.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/private/run_binary.bzl:98]: Hmm. The html comes from upstream, https://github.com/bazelbuild/bazel-skylib/blob/main/rules/run_binary.bzl#L71. Looks good on the docsite https://docs.aspect.build/bazelbuild/bazel-skylib/1.2.1/docs/run_binary_doc_gen.html#run_binary-tool so I'm going to leave it.

File: lib/private/expand_make_vars.bzl [hunk 3]
```diff
@@ -1,30 +1,9 @@
 "Helpers to expand make variables"
```
Comment (alexeagle) [inline lib/private/expand_make_vars.bzl:1]: wow we had so much rules_nodejs legacy crap in here that wasn't really needed??

File: docs/copy_file.md (hunk 2 above)
Comment (gregmagolan) [reply docs/copy_file.md:11]: Sounds reasonable to me

File: lib/private/expand_make_vars.bzl (hunk 3 above)
Comment (gregmagolan) [reply lib/private/expand_make_vars.bzl:1]: Yeah. That was some old code from RNJ. We should flip it in RNJ on the next major as well.

**Pull request merged**